### PR TITLE
Get-DecryptedString support for CNG cert private keys

### PR DESCRIPTION
**Pull request by jason-qq**
Add support for CNG certificate private keys
Kept CSP functionality as before

### CODE REVIEW

**Review by joshnielsen10 — APPROVED** [review]

File: common/Fabric-Install-Utilities.psm1 [hunk 1]
```diff
@@ -580,13 +580,21 @@ function Get-Certificate($certificateThumbprint)
...
         $cleanedEncryptedString = $encryptedString.Replace("!!enc!!:","")
-        $clearTextValue = [System.Text.Encoding]::UTF8.GetString($encryptionCertificate.PrivateKey.Decrypt([System.Convert]::FromBase64String($cleanedEncryptedString), $true))
-        return $clearTextValue
-    }else{
+    } else{
         return $encryptedString
     }
+
+    if ($null -eq $encryptionCertificate.PrivateKey) {
+        $privateKey = [System.Security.Cryptography.X509Certificates.RSACertificateExtensions]::GetRSAPrivateKey($encryptionCertificate)
```
Comment (miketabet) [inline common/Fabric-Install-Utilities.psm1]: Add logging here so we can track whether teh cert is CNG or CSP.

File: common/Fabric-Install-Utilities.psm1 [hunk 2]
```diff
@@ -580,13 +580,21 @@ function Get-Certificate($certificateThumbprint)
...
 
 function Get-DecryptedString($encryptionCertificate, $encryptedString){
-    if($encryptedString.StartsWith("!!enc!!:")){
+    if ($encryptedString.StartsWith("!!enc!!:")) {
         $cleanedEncryptedString = $encryptedString.Replace("!!enc!!:","")
-        $clearTextValue = [System.Text.Encoding]::UTF8.GetString($encryptionCertificate.PrivateKey.Decrypt([System.Convert]::FromBase64String($cleanedEncryptedString), $true))
-        return $clearTextValue
-    }else{
+    } else{
         return $encryptedString
```
Comment (miketabet) [inline common/Fabric-Install-Utilities.psm1:586]: Log here if there's an error.

File: common/Fabric-Install-Utilities.psm1 (hunk 1 above)
Comment (miketabet) [inline common/Fabric-Install-Utilities.psm1]: Try catch on each of these to catch and log errors in readable messages.

**Review by miketabet — CHANGES_REQUESTED** [review]
Good code but needs more logging and error handling to make it better when run during installs.

File: common/Fabric-Install-Utilities.psm1 (hunk 1 above)
Comment (jason-qq) [reply common/Fabric-Install-Utilities.psm1]: Can do

**Pull request merged**